lib/galaxy/model/tool_shed_install/mapping.py

```diff
@@ -1,22 +1,31 @@
+from typing import (
+    Any,
+    TYPE_CHECKING,
+    Union,
+)
+
 from galaxy.model import tool_shed_install as install_model
 from galaxy.model.base import ModelMapping
 from galaxy.model.orm.engine_factory import build_engine
 from galaxy.model.tool_shed_install import mapper_registry
 
+if TYPE_CHECKING:
+    from sqlalchemy.engine import Engine
+
 metadata = mapper_registry.metadata
 
 
-def init(url, engine_options=None, create_tables=False):
+def init(url: str, engine_options: Union[dict[str, Any], None] = None, create_tables: bool = False) -> ModelMapping:
```

Comment:
Why the Union with None instead of just Optional? Not going to prevent the merge on it just curious.

Reply:
`Optional` is soft-deprecated by the Python typing developers, see e.g.:
- https://typing.python.org/en/latest/reference/best_practices.html#shorthand-syntax
- https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/6

So I started using `Union` instead of `Optional` while waiting for when we drop Python 3.9 and we can update to `|` with https://github.com/asottile/pyupgrade?tab=readme-ov-file#pep-604-typing-rewrites